### PR TITLE
SSCS-10222 Gaps Route Splitting

### PR DESCRIPTION
**Pull request by MohammedHMCTS**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10222

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java [hunk 1]
```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.sscs.ccd.domain;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+@Getter
+public enum HearingRoute {
+
+    @JsonEnumDefaultValue
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java:10]: I'm not sure if we want to do this. It should be the default but only when we release S&L

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
+package uk.gov.hmcts.reform.sscs.ccd.domain;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+@Getter
+public enum HearingState {
+
+    @JsonEnumDefaultValue
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java:10]: Same here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java [hunk 3]
```diff
@@ -32,25 +33,28 @@ public RegionalProcessingCenter(@JsonProperty("faxNumber") String faxNumber,
         this.postcode = postcode;
         this.city = city;
         this.email = email;
+        this.hearingRoute = hearingRoute;
     }
 
-    private String faxNumber;
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:37]: Why were these removed?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    @JsonProperty
+    LIST_ASSIST("listAssist"),
+    @JsonProperty
+    GAPS("gaps");
+
+    private final String type;
+
+    HearingRoute(String type) {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java:18]: Optional: Can use lombok `@RequiredArgsConstructor` for this

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    @JsonProperty
+    HEARING_CREATED("hearingCreated");
+
+    private final String state;
+
+    HearingState(String state) {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java:16]: Optional: Can use lombok @RequiredArgsConstructor for this

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
+package uk.gov.hmcts.reform.sscs.ccd.domain;
+
+import com.fasterxml.jackson.annotation.JsonEnumDefaultValue;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.Getter;
+
+@Getter
+public enum HearingRoute {
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java:8]: I don't get why the field 'type' is on this enum as jackson should be able to map to the enum name - but you've probably done it for a reason.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java (hunk 3 above)
Comment (wmarshid) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:37]: The class uses lombok `@value` annotation which makes fields immutable (gives them private access and final keyword) `@value` is shorthand for `final @ToString @EqualsAndHashCode @AllArgsConstructor @FieldDefaults(makeFinal = true, level = AccessLevel.PRIVATE) @Getter`

**Review by wmarshid — APPROVED** [review]
Approved with comments

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java (hunk 3 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/RegionalProcessingCenter.java:37]: Sorry yeah I replied on the last PR that it automatically sets it but closed that PR

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java (hunk 4 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingRoute.java:18]: Yup that is fine will update those

**Review by dattipoeHMCTS — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java [hunk 7]
```diff
@@ -1,24 +1,30 @@
...
 import lombok.Getter;
 
 @Getter
+@AllArgsConstructor
 public enum HearingState {
 
-    @JsonEnumDefaultValue
     @JsonProperty
-    HEARING_CREATED("hearingCreated");
+    CREATE_HEARING("createHearing"),
```
Comment (wmarshid) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/HearingState.java:13]: Minor: Add line breaks between fields

File: src/main/java/uk/gov/hmcts/reform/sscs/service/RegionalProcessingCenterService.java [hunk 8]
```diff
@@ -130,4 +131,11 @@ public static String getFirstHalfOfPostcode(String postcode) {
         return "";
     }
 
+    public HearingRoute getHearingRoute(String region) {
+        return regionalProcessingCenterMap.values().stream()
+            .filter(rpc -> rpc.getName().equalsIgnoreCase(region))
+            .map(RegionalProcessingCenter::getHearingRoute)
+            .findFirst().orElse(HearingRoute.LIST_ASSIST);
```
Comment (dattipoeHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/service/RegionalProcessingCenterService.java:138]: You have an uncovered test case for the default

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/service/RegionalProcessingCenterService.java (hunk 8 above)
Comment (MohammedHMCTS) [reply src/main/java/uk/gov/hmcts/reform/sscs/service/RegionalProcessingCenterService.java:138]: Added

**Review by dattipoeHMCTS — APPROVED** [review]
